src/librustdoc/html/static/main.js

```diff
@@ -90,6 +104,35 @@ if (!DOMTokenList.prototype.remove) {
         return null;
     }
 
+    function getCratesHash() {
+        if (window.hashCode === undefined || window.hashCode === 0) {
+            if (window.crates === undefined || window.crates.length === 0) {
+                window.crates = cloneArray(document.getElementById("crate-search")
+                                                   .getElementsByTagName("option"), 1)
+                                    .map(function(x) {
+                                        return x.value;
+                                    });
+                window.crates.sort();
+            }
+            window.hashCode = window.crates.join(",").hashCode();
```

Comment:
The whole list of crates isn't that long in most cases, and we store it in memory here anyway, so why do we need to compute it's hash? Can't the crate list simply be stored as a string?

Reply:
For the storage, not just as is.